pyathena/sqlalchemy_athena.py

```diff
@@ -232,16 +288,44 @@ def post_create_table(self, table):
                 " in the connection string."
             )
         schema = table.schema if table.schema else raw_connection.schema_name
-        text += "LOCATION '{0}{1}/{2}/'\n".format(location, schema, table.name)
-
-        compression = raw_connection._kwargs.get("compression")
-        if compression:
-            text += "TBLPROPERTIES ('parquet.compress'='{0}')\n".format(
-                compression.upper()
-            )
+        athena_s3_prefix = table.kwargs.get('athena_s3_prefix')
```

Comment:
There is [a specific table attribute for dialect kwargs](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.Table.dialect_kwargs). The fact the that kwarg was not properly named might explain why it was not available here.

```suggestion
        athena_s3_prefix = table.dialect_kwargs.get('awsathena', {}).get('s3_prefix')
```
Not sure if there is a better was to get the dialect name, something like `self.dialect.name` ?